docs/tutorials/FlowTutorial.md

```diff
@@ -0,0 +1,1581 @@
+```
+source init_floorplan2.tcl
+```
+
+View the resulting core utilization of 30 created following floorplan:
+
+![Relative Floorplan](./images/core_util.png)
+
+#### Defining Placement Density
+
+To learn on placement density strategies for `ibex` design, go to
+`OpenROAD-flow-scripts/flow`. Type:
+
+```
+openroad -gui
+```
+
+Enter following commands in `Tcl Commands` section of GUI
+
+```
+read_lef ./platforms/sky130hd/lef/sky130_fd_sc_hd.tlef
+read_lef ./platforms/sky130hd/lef/sky130_fd_sc_hd_merged.lef
+read_def ./results/sky130hd/ibex/base/3_place.def
+```
+![ibex placement density 60](./images/ibex_pl_60.png)
+
+Change `CORE_UTILIZATION` and `PLACE_DENSITY` for the `ibex` design
+`config.mk` as follows.
+
+View `ibex` design `config,mk` [here](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/designs/sky130hd/ibex/config.mk)
+
+```
+export CORE_UTILIZATION = 40
+export CORE_ASPECT_RATIO = 1
+export CORE_MARGIN = 2
+export PLACE_DENSITY = 0.50
+```
+
+Re-run the `ibex` design with below command:
+
+```
+make DESIGN_CONFIG=./designs/sky130hd/ibex/config.mk
+```
+
+View the `ibex` design placement density heat map as shown below:
+
+![ibex placement density 50](./images/ibex_pl_50.png)
+
+So from above GUI understood that change in `CORE_UTILIZATION` from 20
+to 40 and placement density default 0.60 to 0.50 changes standard cell
+placement became widely spread.
+
+### Chip Level IO Pad Placement
+
+In this section, you will generate an I/O pad ring for the coyote design
+using a Tcl script.
+
+ICeWall is a utility to place IO cells around the periphery of a design,
+and associate the IO cells with those present in the netlist of the
+design.
+
+For I/O pad placement using ICeWall refer readme file [here](../../src/pad/README.md)
+
+Refer to the built-in examples [here](../../src/pad/test)
+
+Launch openroad GUI, by typing:
+
+```
+cd ../../src/pad/test/
+openroad -gui
+```
+
+Run [coyote_tc_sky130.tcl](../../src/pad/test/coyote_tc_sky130.tcl) to view ICeWall based IO pad placement.
+
+From the GUI `Tcl commands` prompt run below script:
+
+```
+source coyote_tc_sky130.tcl
+```
+
+View the resulting IO pad ring in GUI as shown below
+
+![coyote pad ring](./images/coyote_pad_ring.png)
+
+### Power Planning And Analysis
+
+In this section, you will use the design `black_parrot` to create a
+power grid and run power analysis.
+
+Pdngen is used for power planning. Refer to the following `README.md` [here](../../src/pdn/README.md)
+
+Refer to the built-in examples [here](../../src/pdn/test)
+
+Launch openroad GUI, by typing:
+
+```
+cd ../../src/pdn/test
+openroad -gui
+```
+
+To run [soc_bsg_black_parrot_nangate45.tcl](../../src/pdn/test/soc_bsg_black_parrot_nangate45.tcl)
+to generate power grid for `black_parrot` design by typing:
+
+```
+source soc_bsg_black_parrot_nangate45.tcl
+```
+
+Refer to the section of the power domain network strategy file [here](../../src/pdn/test/soc_bsg_black_parrot_nangate45/PDN.cfg)

```

Comment:
This is still referencing the PDN config file instead of the correct tcl commands that should be used instead to generate the power grid.